src/GslibIO.jl

```diff
@@ -52,28 +81,46 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
+  spec = parse_legacy(filename)                     
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  # handle missing values
+  replace!(spec.data, na=>NaN)
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # create data dictionary
+  table = (; zip(spec.varnames, eachcol(spec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # read property values
-    X = readdlm(fs)
+  georef(table, domain)
+end
 
-    # handle missing values
-    replace!(X, na=>NaN)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  spec = parse_legacy(filename)
 
-    georef(data, domain)
+  # handle missing values
+  replace!(spec.data, na=>NaN)
+
+  # we need to identify and separate coordinates and actual attributes 
+  # from the parsed variables
+
+  # find the position of each `coordnames` in `dataspec.varnames`
+  coordinds = indexin(spec.varnames, collect(coordnames))
+  if count(something, coordinds) != length(coordnames)
```

Comment:
```suggestion
  if any(isnothing.(coordinds))
```

Reply:
This suggestion does not work. We need to make sure that each coordname is found in varnames, which means that the number of `coordinds` must match to the number of `coordnames`, not just any number > 0.

Reply:
Notice that you flipped the arguments of `indexin`, the order should be `indexin(collect(coordnames), spec.varnames)`. Also this check about the coordinate names being present or not should come earlier as an assertion, right after you parse the file:

```julia
spec = parse_legacy(file)

@assert coordnames \subseteq spec.varnames "invalid coordinate names"
```

Reply:
Can you please refactor so that the assertion happens early, and then this if-else block `count(something...` is removed?

Reply:
Let me double check that the proposed code works

Reply:
All work fine! See next commit

Reply:
I realised that if we check that coordnames are in varnames, calling to indexin should always give us the right return, therefore no need for an extra check and raise an error.